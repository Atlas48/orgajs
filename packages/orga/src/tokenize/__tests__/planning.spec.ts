import tok from './tok'

const options = {
  timezone: 'Pacific/Auckland',
}

describe('tokenize planning', () => {
  it('knows plannings', () => {
    expect(tok('DEADLINE: <2018-01-01 Mon>', options)).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "DEADLINE:",
          "type": "planning.keyword",
          "value": "DEADLINE",
        },
        Object {
          "_text": " <2018-01-01 Mon>",
          "type": "planning.timestamp",
          "value": Object {
            "date": 2017-12-31T11:00:00.000Z,
            "end": undefined,
          },
        },
      ]
    `)
    expect(tok('  DEADLINE: <2018-01-01 Mon>', options)).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "DEADLINE:",
          "type": "planning.keyword",
          "value": "DEADLINE",
        },
        Object {
          "_text": " <2018-01-01 Mon>",
          "type": "planning.timestamp",
          "value": Object {
            "date": 2017-12-31T11:00:00.000Z,
            "end": undefined,
          },
        },
      ]
    `)
    expect(tok(' \tDEADLINE: <2018-01-01 Mon>', options))
      .toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "DEADLINE:",
          "type": "planning.keyword",
          "value": "DEADLINE",
        },
        Object {
          "_text": " <2018-01-01 Mon>",
          "type": "planning.timestamp",
          "value": Object {
            "date": 2017-12-31T11:00:00.000Z,
            "end": undefined,
          },
        },
      ]
    `)
    expect(tok(' \t DEADLINE: <2018-01-01 Mon>', options))
      .toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "DEADLINE:",
          "type": "planning.keyword",
          "value": "DEADLINE",
        },
        Object {
          "_text": " <2018-01-01 Mon>",
          "type": "planning.timestamp",
          "value": Object {
            "date": 2017-12-31T11:00:00.000Z,
            "end": undefined,
          },
        },
      ]
    `)
  })

  it('know multiple plannings', () => {
    expect(
      tok('DEADLINE: <2020-07-03 Fri> SCHEDULED: <2020-07-03 Fri>', options)
    ).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "DEADLINE:",
          "type": "planning.keyword",
          "value": "DEADLINE",
        },
        Object {
          "_text": " <2020-07-03 Fri> ",
          "type": "planning.timestamp",
          "value": Object {
            "date": 2020-07-02T12:00:00.000Z,
            "end": undefined,
          },
        },
        Object {
          "_text": "SCHEDULED:",
          "type": "planning.keyword",
          "value": "SCHEDULED",
        },
        Object {
          "_text": " <2020-07-03 Fri>",
          "type": "planning.timestamp",
          "value": Object {
            "date": 2020-07-02T12:00:00.000Z,
            "end": undefined,
          },
        },
      ]
    `)
  })

  it('knows these are not plannings', () => {
    expect(tok('dEADLINE: <2018-01-01 Mon>', options)).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "dEADLINE: <2018-01-01 Mon>",
          "type": "text.plain",
          "value": "dEADLINE: <2018-01-01 Mon>",
        },
      ]
    `)
  })
})
