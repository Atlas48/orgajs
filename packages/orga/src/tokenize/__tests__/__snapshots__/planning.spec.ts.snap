// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize planning knows plannings 1`] = `
Array [
  Object {
    "keyword": "DEADLINE",
    "position": Object {
      "end": Object {
        "column": 9,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "planning.keyword",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 26,
        "line": 0,
      },
      "start": Object {
        "column": 9,
        "line": 0,
      },
    },
    "timestamp": Object {
      "date": 2017-12-31T11:00:00.000Z,
      "end": undefined,
    },
    "type": "planning.timestamp",
  },
]
`;

exports[`tokenize planning knows plannings 2`] = `
Array [
  Object {
    "keyword": "DEADLINE",
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "planning.keyword",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 28,
        "line": 0,
      },
      "start": Object {
        "column": 11,
        "line": 0,
      },
    },
    "timestamp": Object {
      "date": 2017-12-31T11:00:00.000Z,
      "end": undefined,
    },
    "type": "planning.timestamp",
  },
]
`;

exports[`tokenize planning knows plannings 3`] = `
Array [
  Object {
    "keyword": "DEADLINE",
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "planning.keyword",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 28,
        "line": 0,
      },
      "start": Object {
        "column": 11,
        "line": 0,
      },
    },
    "timestamp": Object {
      "date": 2017-12-31T11:00:00.000Z,
      "end": undefined,
    },
    "type": "planning.timestamp",
  },
]
`;

exports[`tokenize planning knows plannings 4`] = `
Array [
  Object {
    "keyword": "DEADLINE",
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 3,
        "line": 0,
      },
    },
    "type": "planning.keyword",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 29,
        "line": 0,
      },
      "start": Object {
        "column": 12,
        "line": 0,
      },
    },
    "timestamp": Object {
      "date": 2017-12-31T11:00:00.000Z,
      "end": undefined,
    },
    "type": "planning.timestamp",
  },
]
`;

exports[`tokenize planning knows these are not plannings 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 26,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "dEADLINE: <2018-01-01 Mon>",
  },
]
`;
